Add ability to detekt check only on changed files (#49)
RedMadRobot/gradle-infrastructure: README.md

- Configures additional tasks:
    - `detektAll` - Runs Detekt over the whole codebase
    - `detektFormat` - Reformats the whole codebase with Detekt

>     - `detektDiff` - Runs Detekt only on changed files if it enables

Reviewer: ```suggestion
    - `detektDiff` - Runs Detekt only on changed files (see [Enable Detekt checks only on changed files][#enable-detekt-checks-only-on-changed-files])
```

Author: Added.

}
```

### Enable Detekt checks only on changed files

> The plugin provides the ability to check only changed files. You can enable this:

Reviewer: The description should be more detailed.

```suggestion
Plugin `redmadrobot.detekt` adds task `detektDiff` to check the only files changed comparing to the base branch.
To enable this feature, you should specify base branch name:
```

Author: Applied you suggestion.

```kotlin
redmadrobot {
    detekt {
        checkOnlyDiffWithBranch("develop")
    }
}
```

> The plugin then adds a `detektDiff` task that allows you to check only changed files versus the base branch. The modified files are provided by Git.
> > Task `detektDiff` work with rules without **type resolutions**. About **type resolution** you can read [here](https://detekt.github.io/detekt/type-resolution.html).

Reviewer: It is better to leave blank lines between code blocs, callouts, headers, and text. Also, it is useful to break lines before each sentence.  
Also, here text can be shorter.

```suggestion
The plugin then adds a `detektDiff` task that allows you to check only changed files versus the base branch.
The modified files are provided by Git.

> Task `detektDiff` is not compatible with [**type resolution**](https://detekt.github.io/detekt/type-resolution.html).
> It means some of the configured rules will not work.
```
Btw maybe better to place the disclaimer about type resolution under detekt tasks list? Because all of the tasks not work with type resolution.

Author: Formatted text and added a note to the `detekt` section.
